llvm/include/llvm/MC/MCDecoder.h

```diff
@@ -58,6 +59,24 @@ uint64_t fieldFromInstruction(const std::bitset<N> &Insn, unsigned StartBit,
   return ((Insn >> StartBit) & Mask).to_ullong();
 }
 
+template <unsigned StartBit, unsigned NumBits, typename T>
+inline std::enable_if_t<std::is_unsigned_v<T>, T> extractBits(T Val) {
+  static_assert(StartBit + NumBits <= std::numeric_limits<T>::digits);
```

Comment:
this seems incorrect? Shouldn't it be sizeof(T)*CHAR_BIT or something like that?